test/ipns.js

```diff
@@ -7,121 +7,219 @@ const expect = chai.expect
         expect(err).to.not.exist()
-        expect(res).to.equal(ipfsRef)
-        cb()
+        publishAndResolve(jsDaemon, done)
       })
-    },
-    (cb) => resolverDaemon.stop(cb),
-    (cb) => setTimeout(cb, 2000),
-    (cb) => resolverDaemon.cleanup(cb)
-  ], callback)
-}
-
-describe('ipns locally using the same repo across implementations', () => {
-  it('should publish an ipns record to a js daemon and resolve it using the same js daemon', function (done) {
-    this.timeout(120 * 1000)
-    const dir = path.join(os.tmpdir(), hat())
-
-    spawnJsDaemon(dir, (err, jsDaemon) => {
-      expect(err).to.not.exist()
-      publishAndResolve(jsDaemon, done)
     })
-  })
 
-  it('should publish an ipns record to a go daemon and resolve it using the same go daemon', function (done) {
-    this.timeout(160 * 1000)
-    const dir = path.join(os.tmpdir(), hat())
+    it('should publish an ipns record to a go daemon and resolve it using the same go daemon', function (done) {
+      this.timeout(160 * 1000)
+      const dir = path.join(os.tmpdir(), hat())
 
-    spawnGoDaemon(dir, (err, goDaemon) => {
-      expect(err).to.not.exist()
-      publishAndResolve(goDaemon, done)
+      spawnGoDaemon(dir, (err, goDaemon) => {
+        expect(err).to.not.exist()
+        publishAndResolve(goDaemon, done)
+      })
     })
-  })
 
-  it('should publish an ipns record to a js daemon and resolve it using a go daemon through the reuse of the same repo', function (done) {
-    this.timeout(120 * 1000)
-    const dir = path.join(os.tmpdir(), hat())
+    it('should publish an ipns record to a js daemon and resolve it using a go daemon through the reuse of the same repo', function (done) {
+      this.timeout(120 * 1000)
+      const dir = path.join(os.tmpdir(), hat())
+
+      series([
+        (cb) => spawnJsDaemon(dir, cb),
+        (cb) => spawnGoDaemon(dir, cb)
+      ], (err, daemons) => {
+        expect(err).to.not.exist()
+
+        publishAndResolve(daemons[0], daemons[1], done)
+      })
+    })
 
-    series([
-      (cb) => spawnJsDaemon(dir, cb),
-      (cb) => spawnGoDaemon(dir, cb)
-    ], (err, daemons) => {
-      expect(err).to.not.exist()
+    it('should publish an ipns record to a go daemon and resolve it using a js daemon through the reuse of the same repo', function (done) {
+      this.timeout(160 * 1000)
+      const dir = path.join(os.tmpdir(), hat())
 
-      publishAndResolve(daemons[0], daemons[1], done)
+      series([
+        (cb) => spawnGoDaemon(dir, cb),
+        (cb) => spawnJsDaemon(dir, cb)
+      ], (err, daemons) => {
+        expect(err).to.not.exist()
+
+        publishAndResolve(daemons[0], daemons[1], done)
+      })
     })
   })
 
-  it('should publish an ipns record to a go daemon and resolve it using a js daemon through the reuse of the same repo', function (done) {
-    this.timeout(160 * 1000)
-    const dir = path.join(os.tmpdir(), hat())
+  describe('ipns over dht', () => {
```

Comment:
These are very different tests to the repo tests above - I'd pull them out into a spearate file.